HPXMLtoOpenStudio/resources/airflow.rb

```diff
@@ -1605,8 +1605,8 @@ def self.apply_mechanical_ventilation(model, vent_fans_mech, living_ach50, livin
     cfis_cfm_tot = vent_mech_cfis_tot.map { |vent_mech| vent_mech.average_total_unit_flow_rate }.sum(0.0)
 
     # Average preconditioned oa air cfms (only oa, recirculation will be addressed below for all shared systems)
-    oa_cfm_preheat = vent_mech_preheat.map { |vent_mech| vent_mech.average_oa_unit_flow_rate }.sum(0.0)
-    oa_cfm_precool = vent_mech_precool.map { |vent_mech| vent_mech.average_oa_unit_flow_rate }.sum(0.0)
+    oa_cfm_preheat = vent_mech_preheat.map { |vent_mech| vent_mech.average_oa_unit_flow_rate * vent_mech.preheating_fraction_load_served }.sum(0.0)
+    oa_cfm_precool = vent_mech_precool.map { |vent_mech| vent_mech.average_oa_unit_flow_rate * vent_mech.precooling_fraction_load_served }.sum(0.0)
```

Comment:
@shorowit Sorry for bothering again for the hvac sizing. I am a bit concerned about an ERV/HRV fan combined with a preconditioned shared system. Is it assuming that the preconditioning happens after heat/energy recovery? Then I think airflows in hvac sizing should not subtract all the `vent_mech.average_oa_unit_flow_rate` as preconditioned (assuming fraction load served is 1.0), but only the portion after recovery. It looks like there's a portion of OA where the ERV/HRV heat recovery and preconditioning have been both counted in hvac sizing. Does it make sense? Though not sure how much it will impact the results and whether it is a case worth more efforts.

Reply:
So I ran 2 files: 1) A shared HRV w/ preconditioning and 2) the same file without any mechanical ventilation. I then looked at the design loads for these two files (hint: you can see them by running workflow/run_simulation.rb with the --debug flag and then looking at the extra output in run.log). All of the design loads were identical (except for Cool Infil Lat). This seems like the correct result to me, as the infiltration should be the same between a balanced ventilation system and no ventilation system, and the ventilation outdoor air load introduced is being fully met by the preconditioning equipment.

I then ran the same exercise but with a shared supply system w/ preconditioning instead of HRV. In this case, the design loads were not identical, since the infiltration will differ between the supply system and no mech vent system (while the ventilation outdoor air load introduced is presumably still being fully met by the preconditioning equipment).

So I don't have any concerns here unless I missed your point.

Reply:
GitHub isn't letting me upload my test files for some reason so I'll send on Slack.

Reply:
Oh I figured it out. I [only looped non-preconditioned ERV/HRV fans for combined effectiveness](https://github.com/NREL/OpenStudio-HPXML/blob/shared-mech-vent-precond-v3/HPXMLtoOpenStudio/resources/airflow.rb#L1625-L1628), so that preconditioning and heat/energy recovery won't be both fully-counted in a single system. Apparently I've thought about this issue and have avoided it.🤦